components/base/wheeled/wheeled_base.go

```diff
@@ -396,6 +432,20 @@ func CreateWheeledBase(
 		base.right = append(base.right, m)
 	}
 
+	// spawn a new context for sensors so we don't add many background workers
+	// TODO RSDK-2384 something is cancelling base context in the actuating API calls
+	base.sensorCtx, base.sensorDone = context.WithCancel(context.Background())
```

Comment:
this feels super hacky and fragile. this also means that `spinWithMovementSensor` won't respond context cancelation of the original command.

Reply:
I've added calls to cancel context for each API call, but I can't use the contexts from each call to do anything. I only create a longstanding background context here, and close it in `base.Close`